510_Deployment/50_heap.asciidoc

```diff
@@ -116,86 +90,75 @@ $ JAVA_HOME=`/usr/libexec/java_home -v 1.8` java -Xmx32767m -XX:+PrintFlagsFinal
      bool UseCompressedOops   = false
 ----
 
-The morale of the story is that the exact cutoff to leverage compressed oops
-varies from JVM to JVM, so take caution when taking examples from elsewhere and
-be sure to check your system with your configuration and JVM.
+这个例子告诉我们，影响内存指针压缩使用的临界值，
+是会根据 JVM 的不同而变化的。
+所以从其他地方获取的例子，需要谨慎使用，要确认检查操作系统配置和 JVM。
 
-Beginning with Elasticsearch v2.2.0, the startup log will actually tell you if your
-JVM is using compressed OOPs or not.  You'll see a log message like:
+如果使用的是  Elasticsearch v2.2.0，启动日志其实会告诉你 JVM 是否正在使用内存指针压缩。
+你会看到像这样的日志消息：
 
 [source, bash]
 ----
 [2015-12-16 13:53:33,417][INFO ][env] [Illyana Rasputin] heap size [989.8mb], compressed ordinary object pointers [true]
 ----
 
-Which indicates that compressed object pointers are being used.  If they are not,
-the message will say `[false]`.
-
+这表明内存指针压缩正在被使用。如果没有，日志消息会显示 `[false]` 。
 
 [role="pagebreak-before"]
-.I Have a Machine with 1 TB RAM!
+.我有一个 1 TB 内存的机器！
 ****
-The 32 GB line is fairly important.  So what do you do when your machine has a lot
-of memory?  It is becoming increasingly common to see super-servers with 512&#x2013;768 GB
-of RAM.
+这个 32 GB 的分割线是很重要的。那如果你的机器有很大的内存怎么办呢？
+一台有着 512&#x2013;768 GB内存的服务器愈发常见。
 
-First, we would recommend avoiding such large machines (see <<hardware>>).
+首先，我们建议避免使用这样的高配机器（参考 <<hardware>>）。
 
-But if you already have the machines, you have two practical options:
+但是如果你已经有了这样的机器，你有两个可选项：
 
-- Are you doing mostly full-text search?  Consider giving just under 32 GB to Elasticsearch
-and letting Lucene use the rest of memory via the OS filesystem cache.  All that
-memory will cache segments and lead to blisteringly fast full-text search.
+- 你主要做全文检索吗？考虑给 Elasticsearch 不超过 32 GB 的内存，
+让 Lucene 通过操作系统文件缓存来利用余下的内存。那些内存都会用来缓存 segments，带来极速的全文检索。
 
-- Are you doing a lot of sorting/aggregations?  You'll likely want that memory
-in the heap then.  Instead of one node with more than 32 GB of RAM, consider running two or
-more nodes on a single machine.  Still adhere to the 50% rule, though.  So if your
-machine has 128 GB of RAM, run two nodes, each with just under 32 GB.  This means that less
-than 64 GB will be used for heaps, and more than 64 GB will be left over for Lucene.
+- 你需要更多的排序和聚合？你可能会更希望那些那些内存用在堆中。
+你可以考虑一台机器上创建两个或者更多 ES 节点，而不要部署一个使用或者超过 32 GB 内存的节点。
+仍然要坚持 50％ 原则。假设你有个机器有 128 GB 的内存，
+你可以创建两个节点，每个节点内存分配不超过 32 GB。
+也就是说不超过 64 GB 内存给 ES 的堆内存，剩下的超过 64 GB 的内存给 Lucene。
 +
-If you choose this option, set `cluster.routing.allocation.same_shard.host: true`
-in your config.  This will prevent a primary and a replica shard from colocating
-to the same physical machine (since this would remove the benefits of replica high availability).
+如果你选择第二种，你需要配置 `cluster.routing.allocation.same_shard.host: true`。
```

Comment:
已修改

2016-03-07 16:10 GMT+08:00 Medcl <notifications@github.com>:

> In 510_Deployment/50_heap.asciidoc
> <https://github.com/elasticsearch-cn/elasticsearch-definitive-guide/pull/11#discussion_r55171774>
> :
>
> >  +
> > -If you choose this option, set `cluster.routing.allocation.same_shard.host: true`
> > -in your config.  This will prevent a primary and a replica shard from colocating
> > -to the same physical machine (since this would remove the benefits of replica high availability).
> > +如果你选择第二种，你需要配置 `cluster.routing.allocation.same_shard.host: true`。
>
> cluster.routing.allocation.same_shard.host: true 格式好像有点问题
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elasticsearch-cn/elasticsearch-definitive-guide/pull/11/files#r55171774>
> .
>